schemer/__init__.py

```diff
@@ -12,11 +12,12 @@ def __init__(self, contained_type):
 class Schema(object):
     """A Schema encapsulates the structure and constraints of a dict."""
 
-    def __init__(self, doc_spec, strict=True):
+    def __init__(self, doc_spec, strict=True, validators=[]):
```

Comment:
Let's call this `validates` rather than `validators` so that it's consistent with the way we specify validators for fields. 

The idea is that the validation function should be expressed as a predicate so that you can write `validates=[is_a_something_something]`.